File: src/BaseHmiTypes/Projects/HmiScreenDescriptor.cs
namespace BaseHmiTypes.Projects;

public class HmiScreenDescriptor : HmiProjectItemDescriptor
{
    public string? DisplayName { get; set; }

    public HmiScreenDescriptor()
    {
        Kind = HmiProjectItemKind.Screen;
        FolderType = HmiProjectFolderType.Screens;
    }
}

File: typescript/base-hmi-types/src/projects/HmiScreenDescriptor.ts
import { HmiProjectFolderType } from "./HmiProjectFolderType.js";
import { HmiProjectItemDescriptor } from "./HmiProjectItemDescriptor.js";
import { HmiProjectItemKind } from "./HmiProjectItemKind.js";

export class HmiScreenDescriptor extends HmiProjectItemDescriptor {
  displayName?: string;

  constructor() {
    super();
    this.kind = HmiProjectItemKind.Screen;
    this.folderType = HmiProjectFolderType.Screens;
  }
}
